File: 1160-find-words-that-can-be-formed.java
public class 1160-find-words-that-can-be-formed {
    class Solution {
        public int countCharacters(String[] words, String chars) {
            
            String ans = "";
    
            //iterate over words
            for (String word : words) {
    
                //initialize a map for chars
                Map<Character, Integer> charMap = new HashMap<>();
                //create the map with char and its frequencies
                for (var c : chars.toCharArray()){
                    charMap.put(c, charMap.getOrDefault(c, 0) + 1);
                }
    
                //iterate over word
                for (int i=0; i<word.length(); i++){
                    //if word is not in chars map or the value for this char is less than 1
                    if (!charMap.containsKey(word.charAt(i)) || charMap.get(word.charAt(i)) < 1) {break;}
                    //if word is in map and the value is more or equal to 1
                    if (charMap.containsKey(word.charAt(i)) && charMap.get(word.charAt(i)) >= 1){
                        //delete 1 to the value of this char
                        charMap.put(word.charAt(i), charMap.get(word.charAt(i)) - 1);
                        //if this is the last iteration (last letter of this word)
                        if (i == word.length() -1 ){ 
                            ans += word; 
                        }
                    }
                }
            }
            //return len on answer
            return ans.length();
        }
    }
}

File: 1160-find-words-that-can-be-formed.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:

        #string good if can be formed using letters from char (only one)
        #check if each word uses ONLY words from char
        #add words into a var and return len of var

        result = ""

        # map chars = { l: frequency }
        chars_map = {}
        for c in chars:
            chars_map[c] = chars_map.get(c, 0) + 1
        # print(chars_map)
        
        #iterate words
        for word in words:
            #for each word, map letteres and frequency
            word_map = {}
           
            for c in word:
                word_map[c] = word_map.get(c, 0) + 1
            # print(word_map)
            
            temp_word = word
            for k, v in word_map.items():

                if k not in chars_map or v > chars_map[k]:
                    temp_word = ""
                    break
            result+=temp_word
        #return len of result
        return len(result)